### ISSUE TITLE
Failed to run editor

### ISSUE THREAD

**Issue by gilbertbw** [jj-vcs/jj#1966]
## Steps to Reproduce the Problem

1. Add something like this to `.jjconfig`
    ```
    [ui]
    editor = "code -w"
    ```
2. Ensure "code" is on the PATH
2. Run a command that loads the editor, like `jj describe`

## Expected Behavior
jj opens my configured editor.

## Actual Behavior
I get a `Error: Failed to run editor 'code -w'` message

## Specifications

- Platform: Windows 11
- Version: jj 0.8.0-8149ec6df6490ac596f92fdf584d2d666e8c0b6b

**Comment by necauqua** [comment]
`ui.editor` is a command name, so if you want extra args you'd have to write a wrapper script.

Even if the config did accept arguments (which it probably should tbh, so we can keep this issue open for that), it would be `editor = ["code", "=w"]`

**Comment by gilbertbw** [comment]
It looks to me like it does split on spaces to get the arguments https://github.com/martinvonz/jj/blob/bf2af12b3bb4fe5896af4fbef953f7885ca3adb9/src/config.rs#L447

With a config of:
```
[ui]
editor = "code"
```

I still get:
```
Error: Failed to run editor 'code'
```

**Comment by yuja** [comment]
Yes, both `"code -w"` and `["code", "-w"]` are supported.

I don't know if there are Windows-specific issues (like too long `%PATH%`), but maybe you can try the absolute path?

```
[ui]
editor = ['C:\Program Files\...', '-w']
```

**Comment by gilbertbw** [comment]
```
[ui]
editor = ["C:/Users/Gilbert/AppData/Local/Programs/Microsoft VS Code/bin/code", "-w"]
```

has the same issue, but this works:
```
[ui]
editor = ['C:\Users\Gilbert\AppData\Local\Programs\Microsoft VS Code\Code.exe', '-w']
```

**Comment by yuja** [comment]
Is the `bin/code` a real executable (ends with `.exe`) or some wrapper script? Since the command is run without going through `cmd.exe /c`, a wrapper script might not work. I have no idea what can be directly executed on Windows.

**Comment by necauqua** [comment]
oops, ignore me I guess, I just did not see it was parsing `CommandNameAndArgs` from the config and not just a `String`

**Comment by JeppeKlitgaard** [comment]
I am still running into this problem on Windows — it does not appear to correctly resolve `ui.editor = "code"` using the PATH environment.

This is unexpected behaviour given the [documentation](https://docs.jj-vcs.dev/v0.44.0/config/#editor) and the equivalent setting for Git.

As mentioned by @gilbertbw, this works:
```
[ui]
editor = ['C:\Users\Jeppe\AppData\Local\Programs\Microsoft VS Code\Code.exe', '-w']
```

Running `code` in a terminal correctly opens VSCode.

Example:
```
$ jj config edit --user
Editing file: C:\Users\jeppe\AppData\Roaming\jj\config.toml
Error: Failed to run editor 'code'
Caused by: program not found
```

**Comment by PhilipMetzger** [comment]
> Running `code` in a terminal correctly opens VSCode.

Yes, because that invokes `code.cmd`, so if you replace config setting with it should work with no problem.

**Comment by JeppeKlitgaard** [comment]
> > Running `code` in a terminal correctly opens VSCode.
> 
> Yes, because that invokes `code.cmd`, so if you replace config setting with it should work with no problem.

You're right, thanks! `ui.editor = "code.cmd"` seems to work. I would expect this to work without the `.cmd` ending though, similar to Git behaviour.
